### PR TITLE
Rudimentary implementation of pandas.ExtensionArray-based data container

### PR DESCRIPTION
**Pull request by prockenschaub**
## Reference Issues/PRs
Provides a bare-bone implementation of a pandas ExtensionArray based TimeFrame to address the issues with the data container raised in #15.

#### What does this implement/fix? Explain your changes.
This is the cleaned version of the prototype developed here: [https://github.com/sktime/data-container](https://github.com/sktime/data-container) together with @matteogales, @mloning and @fkiraly. 

As opposed to the current implementation, which uses nested pandas series, the new implementation stores timeseries data as a single numpy array in the background while still allowing full integration with pandas. This has the major advantage that the series can be subset or summarised with fast numpy operations and does not have to iterate over individual pandas.Series objects.

Time indices are stored in a second numpy array. 

By subclassing pandas.DataFrame and pandas.Series, the new implementation also allows to store  meta-data.

#### Any other comments?
The current implementation is a proof of concept and so far only allows for equal length series. If the numpy-solution works as intended, we are thinking about using [awkwardarrays ](https://github.com/scikit-hep/awkward-array) as an alternative backend to also store non-equal length time series.

Due to a large amount of recent changes to the code base that were necessary to conform with pandas ExtensionArray interface (see unit tests in subfolder ./tests/), I have refactored a lot of implementation and interfaces. There are currenlty no docstrings with most functions, I will try to add at least basic docstrings as soon as possible.

### CODE REVIEW

**Comment by prockenschaub** [comment]
Note that the current implementation relies on pandas version 1.0.*. This is different from the current sktime requirement of pandas 0.24.*. The version is mostly relevant for the ExtensionArray unit tests provided by pandas to ensure compatibility. 

Given that this was a major relase, I felt it made sense to design the data container based on the new pandas version but let me know if you have strong feelings otherwise.

File: sktime/container/array.py [hunk 1]
```diff
@@ -0,0 +1,850 @@
...
+        return from_list(data)
+    elif isinstance(data, pd.DataFrame):
+        time_index = data.columns.values[np.newaxis, :]
+        time_index = np.repeat(time_index, data.shape[0], axis=0)
+        return TimeArray(data.to_numpy(), time_index)
+    else:
+        raise TypeError("Input must be a pandas Series or DataFrame: {data}")
+
+
+def from_ts(ts):
```
Comment (mloning) [inline sktime/container/array.py:153]: what do you mean by `ts`? Could you add some reference to the doctoring?

**Comment by mloning** [comment]
@matteogales @fkiraly If you also have some time to review the changes or play around with the examples, that would be great! Patrick updated the [wiki entry](https://github.com/alan-turing-institute/sktime/wiki/Time-series-data-container#Extension-arrays) as well

File: sktime/container/array.py (hunk 1 above)
Comment (prockenschaub) [reply sktime/container/array.py:153]: `ts` was meant to be the ts data format as used in the loading utilities [here](https://github.com/alan-turing-institute/sktime/blob/master/sktime/utils/load_data.py). 

Ultimately, we would provide a same range of importing utilities as we do currently. I needed one that allows me to factorise an object (i.e. turn it into a string) and the ts-like format seemed to be the simplest way to do that. `from_ts`/`to_ts` is therefore a very rudimentary implementation of such a loading function for the sole purpose of factorisation. I can add this to the docstring.

**Pull request merged**